presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -552,6 +558,13 @@ public boolean transitionToFinishing()
             return false;
         }
 
+        try {
+            metadata.endQuery(session);
```

Comment:
If we do this only for reading data, then it doesn't matter if `endQuery` fails. We've already read the data. Cleanup failure doesn't affect that or the correctness of the query.

This is similar to the `close()` method of `InputStream` throwing.